src/views/RuleEngine/components/ConnectorDialog.vue

```diff
@@ -5,152 +5,227 @@
-      type: Boolean,
-      required: false,
-      default: () => false,
-    },
-    modelValue: {
-      type: Object,
-      required: false,
-      default: () => ({}),
-    },
-    connType: {
-      type: String,
-      required: false,
-      default: () => 'mqtt',
-    },
+const props = defineProps({
+  edit: {
+    type: Boolean,
+    required: false,
+    default: () => false,
   },
-  setup(props, context) {
-    const tlsParamsDefault = {
-      enable: false,
-      verify: 'verify_none',
-      certfile: '',
-      keyfile: '',
-      cacertfile: '',
-    }
-
-    const { t } = useI18n()
-    const connectorData = ref(
-      props.edit
-        ? {
-            ..._.cloneDeep(props.modelValue),
-            name: props.modelValue.id?.split(':')[1],
-          }
-        : {},
-    )
-    const connectorTLS = ref(
-      props.edit && props.modelValue.ssl
-        ? _.cloneDeep(props.modelValue.ssl)
-        : _.cloneDeep(tlsParamsDefault),
-    )
-    const submitLoading = ref(false)
+  open: {
+    type: Boolean,
+    required: false,
+    default: () => false,
+  },
+  modelValue: {
+    type: Object,
+    required: false,
+    default: () => ({}),
+  },
+  connType: {
+    type: String,
+    required: false,
+    default: () => 'mqtt',
+  },
+})
+const emit = defineEmits(['update:open', 'finish'])
 
-    const visible = computed({
-      get() {
-        return props.open
-      },
-      set(value) {
-        context.emit('update:open', value)
-      },
-    })
+const tlsParamsDefault = {
+  enable: false,
+  verify: 'verify_none',
+  certfile: '',
+  keyfile: '',
+  cacertfile: '',
+}
 
-    const submitConnector = async (isEdit) => {
-      let res
-      submitLoading.value = true
-      const data = {
-        ...connectorData.value,
-        ssl: { ...connectorTLS.value },
-        type: props.connType,
-      }
-      if (isEdit) {
-        const id = connectorData.value.id
-        // Reflect.deleteProperty(data, "name");
-        Reflect.deleteProperty(data, 'id')
-        Reflect.deleteProperty(data, 'type')
-        Reflect.deleteProperty(data, 'num_of_bridges')
-        res = await updateConnector(id, data).catch(() => {})
-      } else {
-        res = await createConnector(data).catch(() => {})
+const { t } = useI18n()
+const { connectorTypeOptions } = useConnectorTypeValue()
+const connectorData = ref(
+  props.edit
+    ? {
+        ..._.cloneDeep(props.modelValue),
+        // name: props.modelValue.id?.split(':')[1],
       }
-      if (res) {
-        visible.value = false
-        context.emit('finish', true, res)
-        if (!isEdit) {
-          M({ type: 'success', message: t('Base.createSuccess') })
-        } else {
-          M({ type: 'success', message: t('Base.updateSuccess') })
-        }
-      }
-      submitLoading.value = false
-    }
+    : {},
+)
+const connectorTLS = ref(
+  props.edit && props.modelValue.ssl
+    ? _.cloneDeep(props.modelValue.ssl)
+    : _.cloneDeep(tlsParamsDefault),
+)
```

Comment:
Isn't it a bit complicated to include expressions as parameters here?

Reply:
Actually, this is historical code.. But when I want to try to modify it, I found a bug in this logic, tks

Reply:
No bug, just need to optimize the logic 😵